internal/rpcserver/rpcserver.go

```diff
@@ -3690,14 +3705,21 @@ func handleGetWorkRequest(s *Server) (interface{}, error) {
 		templateNtfn := <-templateSub.C()
```

Comment:
I believe this receive also needs to select across the `ctx` as well, otherwise during the initial header sync, the first call to getWork gets stuck here in case of an early context cancellation.

Reply:
Nice catch!  I originally had that along with a timeout too, but then decided that having a timeout there was counter to the goal and ended up removing the context too which I shouldn't have.

Updated.